### PR TITLE
Initial setup for development

### PR DESCRIPTION
**Pull request by ahie**
Basic TypeScript + React setup following the structure of [berth-reservations-ui](https://github.com/city-of-helsinki/ahti-admin-ui) and [berth-reservations-admin](https://github.com/city-of-helsinki/berth-reservations-admin).

### CODE REVIEW

File: tslint.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        ["Symbol", "Avoid using the `Symbol` type. Did you mean `symbol`?"]
+      ]
+    },
+    "prefer-array-literal": {
+      "options": { "allow-type-parameters": true }
+    },
+    "no-console": { "severity": "warning" }
+  },
+  "rulesDirectory": []
+}
```
Comment (Nizar-Rahme) [inline tslint.json]: `tslint` is getting [deprecated](https://github.com/palantir/tslint#tslint) in favor of `eslint`; we recently set up new projects like [berth-reservation-admin](https://github.com/City-of-Helsinki/berth-reservations-admin) and [kukku-ui](https://github.com/City-of-Helsinki/kukkuu-ui/) if you'd like to use them as references.

File: src/domain/app/__tests__/App.test.tsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import App from '../App';
+import { shallow } from 'enzyme';
+
+describe('App', () => {
+  it('renders without crashing', () => {
+    expect(shallow(<App />)).toMatchSnapshot();
```
Comment (Nizar-Rahme) [inline src/domain/app/__tests__/App.test.tsx]: The snapshot here doesn't really reveal much about the component, it took a snapshot for the shallow wrapper itself.
```suggestion
    expect(shallow(<App />).html()).toMatchSnapshot();
```

File: src/domain/app/App.css [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  font-size: calc(10px + 2vmin);
+  color: white;
+}
+
+.App-link {
+  color: #09d3ac;
+}
```
Comment (Nizar-Rahme) [inline src/domain/app/App.css]: In other projects, we're using `SASS` modules, I wonder whether you'd like to consider it as well.

**Review by Nizar-Rahme — CHANGES_REQUESTED** [review]

File: src/domain/app/App.css (hunk 3 above)
Comment (ahie) [reply src/domain/app/App.css]: Yes that was the plan. Updated in the latest commits.

File: src/domain/app/__tests__/App.test.tsx (hunk 2 above)
Comment (ahie) [reply src/domain/app/__tests__/App.test.tsx]: Fixed.

File: tslint.json (hunk 1 above)
Comment (ahie) [reply tslint.json]: Alright, will change this to `eslint`. 👍

Comment (ahie) [reply tslint.json]: Done.

**Pull request merged**